drivers/hmis_external_apis/app/models/hmis_external_apis/ac_hmis/importers/loaders/esg_funding_assistance_loader.rb

```diff
@@ -0,0 +1,108 @@
+  class EsgFundingAssistanceLoader < SingleFileLoader
+    def perform
+      records = build_records
+      # destroy existing records and re-import
+      model_class.where(data_source: data_source).destroy_all if clobber
+
+      # can't do bulk insert here since polymorphic CDE's don't seem to work
+      # and bulk-insert returned ids are not ordered
+      without_paper_trail do
+        records.each { |record| record.save!(validate: false) }
+      end
+      Rails.logger.info "#{self.class.name} inserted: #{records.size} records into #{model_class.table_name}"
+    end
+
+    protected
+
+    def filename
+      'ESGFundingAssistance.csv'
+    end
+
+    def build_records
+      personal_id_by_enrollment_id = Hmis::Hud::Enrollment
+        .where(data_source: data_source)
+        .pluck(:enrollment_id, :personal_id)
+        .to_h
+
+      rows.map do |row|
+        enrollment_id = row_value(row, field: 'ENROLLMENTID')
+        record = model_class.new(
+          EnrollmentID: enrollment_id,
+          CustomServiceID: Hmis::Hud::Base.generate_uuid,
+          FAStartDate: parse_date(row_value(row, field: 'PAYMENTSTARTDATE')),
+          FAEndDate: parse_date(row_value(row, field: 'PAYMENTENDDATE', required: false)),
+          FAAmount: row_value(row, field: 'AMOUNT', required: false),
+          DateCreated: parse_date(row_value(row, field: 'DATECREATED')),
+          DateUpdated: parse_date(row_value(row, field: 'DATEUPDATED')),
+          custom_service_type_id: custom_service_type.id,
+          PersonalID: personal_id_by_enrollment_id.fetch(enrollment_id),
+          DateProvided: today, # FIXME - this isn't right?
+          UserID: row_value(row, field: 'USERID', required: false) || system_user_id,
+          data_source_id: data_source.id,
+        )
+        cde_attrs(row).each do |attr|
+          record.custom_data_elements.build(attr)
+        end
+        record
+      end
+    end
+
+    def cde_attrs(row)
+      [
+        # funding source is supposed to be required but data has missing values
+        [funding_source(row), :funding_source],
+        [row_value(row, field: 'PAYMENTTYPE', required: false), :payment_type],
+      ].map do |value, definition_key|
+        next unless value
+
+        definition = cde_definition(owner_type: model_class.name, key: definition_key)
+        {
+          owner_type: model_class.name,
+          data_element_definition_id: definition.id,
+          value_string: value,
+          DateCreated: parse_date(row_value(row, field: 'DATECREATED')),
+          DateUpdated: parse_date(row_value(row, field: 'DATEUPDATED')),
+          UserID: row_value(row, field: 'USERID', required: false) || system_user_id,
+          data_source_id: data_source.id,
+        }
+      end.compact
+    end
+
+    FUNDING_SOURCE_MAP = {
+      'Allegheny County' => 'Allegheny County ESG',
+      'City of Pittsburgh' => 'City of Pittsburgh ESG',
+      'State of Pennsylvania' => 'State of Pennsylvania ESG',
```

Comment:
Did the stars get stripped out somewhere? In the initial file there were these values
```
 '**Allegheny County**' => 'Allegheny County ESG',
  '**City of Pittsburgh**' => 'City of Pittsburgh ESG',
  '**State of Pennsylvania**' => 'State of Pennsylvania ESG',
```

Reply:
Oh oops, I forgot the asterisks were actually in the data. Fixed now